compiler/rustc_borrowck/src/diagnostics/mod.rs

```diff
@@ -1075,12 +1075,23 @@ impl<'cx, 'tcx> MirBorrowckCtxt<'cx, 'tcx> {
                         let suggest = match tcx.get_diagnostic_item(sym::IntoIterator) {
                             Some(def_id) => {
                                 let infcx = self.infcx.tcx.infer_ctxt().build();
-                                type_known_to_meet_bound_modulo_regions(
+                                pred_known_to_hold_modulo_regions(
                                     &infcx,
                                     self.param_env,
-                                    tcx.mk_imm_ref(tcx.lifetimes.re_erased, tcx.erase_regions(ty)),
-                                    def_id,
-                                    DUMMY_SP,
+                                    ty::Binder::dummy(
+                                        tcx.mk_trait_ref(
+                                            def_id,
+                                            [tcx.mk_imm_ref(
+                                                tcx.lifetimes.re_erased,
+                                                tcx.erase_regions(ty),
+                                            )
+                                            .into()]
+                                            .into_iter()
+                                            .chain(tcx.host_effect()),
```

Comment:
Hmm... I do get the naming confusing between this not returning the effect kind, but a value of the host effect... I'll try out a few namings

Reply:
> why are u using `mk_trait_ref` instead of `mk_trait_ref_with_effect` here?

for diagnostics I opted to just always fall back to host-on. Many times we don't have any context information available anyway